gcloud/bigquery/dataset.py

```diff
@@ -212,12 +303,22 @@ def _set_properties(self, api_response):
         """
         self._properties.clear()
         cleaned = api_response.copy()
+        access = cleaned.pop('access', ())
+        self.access_grants = self._parse_access_grants(access)
         if 'creationTime' in cleaned:
             cleaned['creationTime'] = float(cleaned['creationTime'])
         if 'lastModifiedTime' in cleaned:
             cleaned['lastModifiedTime'] = float(cleaned['lastModifiedTime'])
         self._properties.update(cleaned)
 
+    def _build_access_resource(self):
```

Comment:
This method builds a resource representation for the whole list, not a single grant.  There is only one line specific to an individual grant, so I just inlined it.